### DISCUSSION TITLE
How to Make Real-Time Form Validation Accessible Without Disrupting User Experience?

### DISCUSSION THREAD

**Discussion by ajoseph0** [CivicActions/accessibility#795] [Q&A]
How can I make dynamic forms validation messages accessible to users that rely on screen readers, while ensuring they are non-intrusive for other users.

**Comment by mgifford** [comment]
What have you tried? ARIA Live would be something that shouldn't affect anyone other than assistive technology users (possibly exclusively screen reader users). 

Take a look at: 
  https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA19

**Comment by maggiewachs** [comment]
@ajoseph0, that's a great question. This article covers a lot of ground on validation, including ways to make informational and error messaging useful for folks on screen readers: https://www.smashingmagazine.com/2023/02/guide-accessible-form-validation/
